### PR TITLE
JWT & Bugfixes

### PR DESCRIPTION
**Pull request by criticic**
Issue: #12 
#### Changes proposed in this pull request and/or Screenshots of changes:

# JWT
- Updated JWT secretKey to be stored in an .env file, and modified the corresponding .env.example
- modified hardcoded instances of the jwt key
- send jwt on both login & signup
- send a new jwt token if user creates a room with is creator true

# Bugs
- added a check to verify if user owns room, while adding questions
- also updated to make sure if some person already having a isCreator=true, is added to some other room, then his isCreator is set to false

# Documentation
- added all missing routes data in the api documentation
- added updated tree in readme

### CODE REVIEW

File: src/controllers/userAuth.controllers.js [hunk 1]
```diff
@@ -64,11 +65,18 @@ async function loginUser(req, res) {
             return response_401(res, 'Unauthorized User');
         }
 
+        const token = jwt.sign({
```
Comment (bhupesh98) [inline src/controllers/userAuth.controllers.js:79]: jwt signing should also be present at register function.

**Review by bhupesh98 — COMMENTED** [review]
fixes in room controller and jwt token upgrade on creation of room was also expected.

File: src/controllers/userAuth.controllers.js (hunk 1 above)
Comment (criticic) [reply src/controllers/userAuth.controllers.js:79]: oh okay wait a min

**Comment by bhupesh98** [comment]
While the issue is competitive, so after finalising we are giving all those participants partial points which will be manually assigned to you as a token of appreciation.

Closing this PR as we already accepted the closest one.

**Pull request closed without merging**